### PR TITLE
Feature/ssm service versioning

### PR DESCRIPTION
**Pull request by agentiqd**

### CODE REVIEW

**Comment by agentiqd** [comment]
Screen Shot 2023-01-18 at 8 06 37 PM

**Comment by agentiqd** [comment]
Screen Shot 2023-01-18 at 8 07 38 PM

before merge we need to rollout out per service, per branch params

**Comment by agentiqd** [comment]
let's just start them all at 0.1.1 🤷‍♂️

**Comment by agentiqd** [comment]
Screen Shot 2023-01-18 at 8 34 16 PM
also added latest, img version, and sha tags

File: .github/workflows/docker-build-deploy.yaml [hunk 1]
```diff
@@ -65,12 +66,38 @@ jobs:
...
+          AWS_REGION: us-east-1
+        run: |
+          export AWS_SSM_IMAGE_VERSION_PATH="/aiq/${GITHUB_REPOSITORY#*/}/${{ inputs.branch }}/version"
+          echo $AWS_SSM_IMAGE_VERSION_PATH
+
+          # Pull down current package version from SSM
+          export IMAGE_SEMVER=$(aws ssm get-parameter --name $AWS_SSM_IMAGE_VERSION_PATH --no-with-decryption --output text --query Parameter.Value)
+          echo "[INFO] - current version $IMAGE_SEMVER"
+
+          export MAJOR=${IMAGE_SEMVER%%.*}
```
Comment (BlaiseAgentIQ) [inline .github/workflows/docker-build-deploy.yaml:81]: This is real slick! Is that a bash templating or something specific to ghactions?

**Review by BlaiseAgentIQ — APPROVED** [review]

File: .github/workflows/docker-build-deploy.yaml (hunk 1 above)
Comment (agentiqd) [reply .github/workflows/docker-build-deploy.yaml:81]: Yeah it's bash's parameter expansion feature. Works pretty well for dissecting values. We also use it to pull the branch name in a few spots.

Pretty good summary and cheatsheet for using it.
https://d.cxcore.net/Bash/bash-parameter-expansion-cheatsheet.pdf

**Pull request merged**